lib/git.zsh

```diff
@@ -12,11 +12,53 @@ function __git_prompt_git() {
 # Outputs current branch info in prompt format
 function git_prompt_info() {
   local ref
-  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
+  local branches
+  local info
+  local deatchedtag
+
+  # If oh-my-zsh.hide-status equals to '1', get out.
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" == "1" ]]; then
+    return 0
+  fi
+
+  # If we are on a folder not tracked by git, get out.
+  if [[ "$(__git_prompt_git rev-parse --git-dir 2> /dev/null)" == "" ]]; then
+    return 0
+  fi
+
+  # If oh-my-zsh.classic-info is set, use classic info output of branch only
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.classic-info 2>/dev/null)" == "1" ]]; then
     ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
     ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) || return 0
     echo "$ZSH_THEME_GIT_PROMPT_PREFIX${ref#refs/heads/}$(parse_git_dirty)$ZSH_THEME_GIT_PROMPT_SUFFIX"
+    return 0
+  fi
+
+  branches=$(__git_prompt_git branch --list -a 2> /dev/null)
+  if [[ "${branches}" == "" && "$?" == "0" ]]; then
+      # Look for initialized repos that have zero commits
+      ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
+      ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null)
+      info="<initialized>/${ref#refs/heads/}"
+  elif branch=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null); then
+      # Show upstream remote and current branch
+      ref=$(__git_prompt_git for-each-ref --format='%(upstream:short)' ${branch} 2> /dev/null) || return 0
+      info="${ref#refs/heads/}"
+      if [[ "${info}" == "" ]]; then
+          # Handle for no remote branch
+          info="<no-remote>/$(__git_prompt_git rev-parse --abbrev-ref HEAD)" || return 0
```

Comment:
You don't need to specify there's no remote, just print the information you have. If there's no remote displayed, it's already known that there's no remote.

Reply:
The idea here is to really differentiate once people enable the enhanced git prompt info.

See this: https://github.com/ohmyzsh/ohmyzsh/pull/9188/commits/b787c278b482220854500167301b40693238d929#diff-b67797c4a19e91a6eebdd5b97df83dff900a5ea4cacd63d1df8abbe86839c663L41-L46